src/utils/extractPaginationDetails.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  Operation,
+  SchemaResponse,
+  SchemaType,
+  ObjectSchema
+} from "@azure-tools/codemodel";
+import { isEqual } from "lodash";
+import { PaginationDetails } from "../models/operationDetails";
+import { getLanguageMetadata } from "./languageHelpers";
+import { getTypeForSchema, isSchemaResponse } from "./schemaHelpers";
+import { TypeDetails } from "../models/modelDetails";
+
+interface PaginationExtension {
+  /**
+   * The name of the field in the response that can be paged over.
+   */
+  itemName?: string;
+  /**
+   * Name of the field containing the nextLink value.
+   * An empty object indicates a null value and that all results
+   * are returned in a single page.
+   */
+  nextLinkName?: string | {};
```

Comment:
I'm not 100% convinced that it can actually be undefined so I'll validate that. In the code model we get from autorest, if `nextLinkName` is specified as `null` in the `x-ms-pageable` extension, I see we get back `{}` instead of `null` for that field. It's supposed to be a required field so probably shouldn't ever be `undefined`.